falcon/api.py

```diff
@@ -206,7 +206,11 @@ def __call__(self, env, start_response):
                     # process_response when no error_handler is given
                     # and for whatever exception. If an HTTPError is raised
                     # remaining process_response will be executed later.
-                    self._call_resp_mw(middleware_stack, req, resp, resource)
+
+                    # NOTE(joarley): The condition will prevent that process_reponse
+                    # is called before self._compose_error_response on HTTPErrors
+                    if not isinstance(ex, HTTPError):
```

Comment:
You're right. I see the problem. Wouldn't be better if each `_call_req_mw`, `_call_rsrc_mw` and `_call_resp_mw` had their own try/except block ?